### PR TITLE
TEL-32, TEL-33 Fix boolean values conversion and extra \" escaping in metrics

### PR DESCRIPTION
**Pull request by maxkondr**
In addition an extra `\"` escaping was removed from metrics on top-level of JSON structure

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -111,14 +108,34 @@ func parseMetricsFile(path string) (*File, error) {
...
 	for k, v := range tmpMetrics {
-		s, err := json.Marshal(v)
-		if err != nil {
-			l.Errorw("error during marshalling metric value to JSON, skipping",
-				zap.Any("value", v),
-				zap.Error(err))
-			return nil, err
+		switch reflect.TypeOf(v).Kind() { //nolint:exhaustive
+		case reflect.String:
+			if vs, ok := v.(string); ok {
```
Comment (artemgavrilov) [inline metrics/metrics.go:113]: Does it make sense to check `ok` in type assertion if you already checked type with reflection? Also you can do the same without reflection at all: https://github.com/percona/platform/blob/55498ecf2d969de75ce38cdc1f234d9fbdf5f620/pkg/starlark/values.go#L20

File: metrics/metrics.go [hunk 2]
```diff
@@ -111,14 +108,34 @@ func parseMetricsFile(path string) (*File, error) {
...
+				// handle special case when "true/false" are written as string
+				if strings.ToLower(vs) == "true" {
```
Comment (artemgavrilov) [inline metrics/metrics.go:115]: Not sure, but probably it makes sense to consider `strconv.ParseBool` here

File: metrics/metrics.go [hunk 3]
```diff
@@ -111,14 +108,34 @@ func parseMetricsFile(path string) (*File, error) {
...
+				metrics[k] = "1"
+			} else {
+				metrics[k] = "0"
+			}
+		default:
+			s, err := json.Marshal(v)
+			if err != nil {
+				l.Errorw("error during marshalling metric value to JSON, skipping",
+					zap.Any("key", k), zap.Any("value", v), zap.Error(err))
+				return nil, err
```
Comment (artemgavrilov) [inline metrics/metrics.go:135]: It looks like you are not skipping bad metrics, but interrupt whole processing, no?

File: metrics/metrics.go (hunk 1 above)
Comment (maxkondr) [reply metrics/metrics.go]: done

File: metrics/metrics.go (hunk 2 above)
Comment (maxkondr) [reply metrics/metrics.go]: done

File: metrics/metrics.go (hunk 3 above)
Comment (maxkondr) [reply metrics/metrics.go]: right you are, it's better to skip it

**Review by artemgavrilov — APPROVED** [review]

**Pull request merged**